integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceIntegrationTest.java

```diff
@@ -169,6 +176,45 @@ public void shouldCreateNewTaskAsJson() throws Exception {
 		assertThat(newTask.getIntent(), equalTo(task.getIntent()));
 		assertThat(task.getAuthoredOn(), equalTo(task.getAuthoredOn()));
 		assertThat(task.getLastModified(), equalTo(task.getLastModified()));
+		assertThat(newTask.getOutput(), hasSize(4));
+		assertThat(newTask.getInput(), hasSize(4));
+		for (Task.TaskOutputComponent output : newTask.getOutput()) {
+			if (output.getValue() instanceof Reference) {
+				assertThat(((Reference) output.getValue()).getReference(),
+				    equalTo("DiagnosticReport/9b6f11dd-55d2-4ff6-8ec2-73f6ad1b759e"));
+				
+			} else if (output.getValue() instanceof DateTimeType) {
+				DateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+				String dateResult = sdf.format(((DateTimeType) output.getValue()).getValue());
+				assertThat(dateResult, equalTo("1905-08-23"));
+				
+			} else if (output.getValue() instanceof DecimalType) {
+				assertThat(((DecimalType) output.getValue()).getValueAsNumber().doubleValue(), equalTo(37.38));
+				
+			} else if (output.getValue() instanceof StringType) {
+				assertThat(output.getValue().toString(), equalTo("Blood Pressure"));
+				
+			}
+		}
+		
+		for (Task.ParameterComponent input : newTask.getInput()) {
+			if (input.getValue() instanceof Reference) {
+				assertThat(((Reference) input.getValue()).getReference(),
+				    equalTo("DiagnosticReport/dbbd9f60-b963-4987-9ac6-ed7d9906bd82"));
+				
+			} else if (input.getValue() instanceof DateTimeType) {
+				DateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+				String dateResult = sdf.format(((DateTimeType) input.getValue()).getValue());
+				assertThat(dateResult, equalTo("1905-08-23"));
+				
+			} else if (input.getValue() instanceof DecimalType) {
+				assertThat(((DecimalType) input.getValue()).getValueAsNumber().doubleValue(), equalTo(37.38));
+				
+			} else if (input.getValue() instanceof StringType) {
+				assertThat(input.getValue().toString(), equalTo("Test code"));
+				
+			}
+		}
 	}
 	
 	@Test
```

Comment:
We should also test that the API is able to create these from the JSON and XML successfully (and it would be good to use the XML format as well as the JSON format).

Reply:
done